src/sidecar/connections/watcher.test.ts

```diff
@@ -112,7 +112,7 @@ describe("sidecar/connections/watcher.ts waitForConnectionToBeStable()", () => {
     };
 
     // As if had been just sent from sidecar.
-    connectionStateWatcher.handleConnectionUpdateEvent(websocketMessage);
+    await connectionStateWatcher.handleConnectionUpdateEvent(websocketMessage);
```

Comment:
This was the main offender in the file, but then required the test-callers to await its containing function